api/circulating-supply.ts

```diff
@@ -0,0 +1,35 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node'
+import { ethers, BigNumber } from 'ethers'
+import dotenv from 'dotenv'
+import ERC20 from '../abis/ERC20.json'
+
+dotenv.config()
+
+export default async (request: VercelRequest, response: VercelResponse) => {
+  const provider = ethers.getDefaultProvider(1, {
+    alchemy: process.env.ALCHEMY_ID,
+    infura: process.env.INFURA_ID,
+    etherscan: process.env.ETHERSCAN_ID,
+    quorum: 1,
+  })
+
+  const signer = new ethers.Wallet(process.env.PRIVATE_KEY as string, provider)
```

Comment:
Why do we need a signer if we are going to use the interface of the contract only for read calls?

Reply:
I thought the `Contract` call requires as a signer but looking better we can just pass it a provider.